RecoLocalFastTime/FTLCommonAlgos/plugins/BTLUncalibRecHitAlgo.cc

```diff
@@ -69,6 +75,9 @@ FTLUncalibratedRecHit BTLUncalibRecHitAlgo::makeRecHit(const BTLDataFrame& dataF
     time.second -= timeCorr_p0_ * pow(amplitude.second, timeCorr_p1_) + timeCorr_p2_;
     flag |= (0x1 << 1);
   }
+  //calculate the position
+  position.first = 0.5 * (c_LYSO_ * (time.first - time.second) + bar_length_);   //distance to left
+  position.second = 0.5 * (bar_length_ - c_LYSO_ * (time.first - time.second));  //distance to right
```

Comment:
Hi,
Yeah I think the relationship cannot be relaxed so I should make that change to save space.
But what do you mean by "save `bar_length_` in the hit"? Is that mean we save another value like time and position in the RecHits class? If so, we are still saving two values (`distance_to_left` and `bar_length_`) in every RecHit, so the space every RecHit takes should be about the same right?
Or I guess you mean `bar_length_` should be some constant that we can always extract from geometry so we can just save one quantity(distance_to_left)?

Thanks.

Reply:
I implied that constants will compress to ~nothing when persisted on disk with compression (default)

Reply:
Hi,
So I should add `const float bar_length_` and `float distance_to_left` in the RecHit and UncalibreatedRecHit class then remove the std::pair of the position. Is that correct?

Thanks.

Reply:
sounds good

Reply:
Hi,
I am thinking about whether I can just store the distance from the center of the bar to the RecHit and use positive or negative to represent the direction. Since what is needed in the clusterizer is the adjustment of the LocalPoint of the RecHit based on our calculated position information and position error. So I think in that way we don't even need to stire the length of the bar.
Does that sound reasonable?
Thanks.

Reply:
even better